aot/src/main/java/com/dylibso/chicory/aot/AotCompiler.java

```diff
@@ -710,7 +710,7 @@ private void compileFunction(
                         break;
                     }
                     // collect unique target labels
-                    Map<Integer, Label> targets = new HashMap<>();
+                    Map<Integer, Label> targets = new TreeMap<>();
```

Comment:
It is sparse and we don't know how big it will be. For example:
```
br_table 1 (;@0;) 1 (;@0;) 1 (;@0;)
br_table 0 (;@2;) 1 (;@1;) 0 (;@2;) 1 (;@1;) 0 (;@2;)
```
In the first example, there is only one unique label, but it's not 0. In the second example, there are only two unique labels, but we don't know that up front.

This could also use `LinkedHashMap`. What matters is that we have a consistent iteration order, so that we generate code deterministically.

Reply:
thanks for the detail :+1: